aptos-move/block-executor/src/scheduler.rs

```diff
@@ -570,6 +571,52 @@ impl Scheduler {
         SchedulerTask::NoTask
     }
 
+    pub fn finish_execution_during_commit(&self, txn_idx: TxnIndex, revalidate_suffix: bool) {
+        // Note: It is preferable to hold the validation lock throughout the finish_execution,
+        // in particular before updating execution status. The point was that we don't want
+        // any validation to come before the validation status is correspondingly updated.
+        // It may be possible to make work more granular, but shouldn't make performance
+        // difference and like this correctness argument is much easier to see, in fact also
+        // the reason why we grab write lock directly, and never release it during the whole function.
+        // So even validation status readers have to wait if they somehow end up at the same index.
+        let _validation_status = self.txn_status[txn_idx as usize].1.write();
+
+        let txn_deps: Vec<TxnIndex> = {
+            let mut stored_deps = self.txn_dependency[txn_idx as usize].lock();
+            // Holding the lock, take dependency vector.
+            std::mem::take(&mut stored_deps)
+        };
+
+        // Mark dependencies as resolved and find the minimum index among them.
+        let min_dep = txn_deps
+            .into_iter()
+            .map(|dep| {
+                // Mark the status of dependencies as 'Ready' since dependency on
+                // transaction txn_idx is now resolved.
+                self.resume(dep);
+
+                dep
+            })
+            .min();
+        if let Some(execution_target_idx) = min_dep {
+            // Decrease the execution index as necessary to ensure resolved dependencies
+            // get a chance to be re-executed.
+            self.execution_idx
+                .fetch_min(execution_target_idx, Ordering::SeqCst);
+        }
+
+        let (cur_val_idx, _cur_wave) =
```

Comment:
all this below, we should be able to replace with self.decrease_validation_idx(txn_idx + 1);
(and as mentioned in other comments, we shouldn't need finish_abort_during_commit nor the lock above).